pkg/operator/options/suite_test.go

```diff
@@ -0,0 +1,265 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	. "knative.dev/pkg/logging/testing"
+
+	coreoptions "github.com/aws/karpenter-core/pkg/operator/options"
+	"github.com/aws/karpenter/pkg/apis/settings"
+	"github.com/aws/karpenter/pkg/operator/options"
+	"github.com/aws/karpenter/pkg/test"
+)
+
+var ctx context.Context
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Options")
+}
+
+var _ = Describe("Options", func() {
+	var envState map[string]string
+	var environmentVariables = []string{
+		"ASSUME_ROLE_ARN",
+		"ASSUME_ROLE_DURATION",
+		"CLUSTER_CA_BUNDLE",
+		"CLUSTER_NAME",
+		"CLUSTER_ENDPOINT",
+		"ISOLATED_VPC",
+		"VM_MOMORY_OVERHEAD_PERCENT",
+		"INTERRUPTION_QUEUE",
+		"RESERVED_ENIS",
+	}
+
+	var fs *coreoptions.FlagSet
+	var opts *options.Options
+
+	BeforeEach(func() {
+		envState = map[string]string{}
+		for _, ev := range environmentVariables {
+			val, ok := os.LookupEnv(ev)
+			if ok {
+				envState[ev] = val
+			}
+			os.Unsetenv(ev)
+		}
+		fs = &coreoptions.FlagSet{
+			FlagSet: flag.NewFlagSet("karpenter", flag.ContinueOnError),
+		}
+		opts = &options.Options{}
+		opts.AddFlags(fs)
+
+		// Inject default settings
+		var err error
+		ctx, err = (&settings.Settings{}).Inject(ctx, nil)
+		Expect(err).To(BeNil())
+	})
+
+	AfterEach(func() {
+		for _, ev := range environmentVariables {
+			os.Unsetenv(ev)
+		}
+		for ev, val := range envState {
+			os.Setenv(ev, val)
+		}
+	})
+
+	Context("Merging", func() {
+		It("shouldn't overwrite options when all are set", func() {
+			err := opts.Parse(
+				fs,
+				"--assume-role-arn", "options-cluster-role",
+				"--assume-role-duration", "20m",
+				"--cluster-ca-bundle", "options-bundle",
+				"--cluster-name", "options-cluster",
+				"--cluster-endpoint", "https://options-cluster",
+				"--isolated-vpc",
+				"--vm-memory-overhead-percent", "0.1",
+				"--interruption-queue", "options-cluster",
+				"--reserved-enis", "10",
+			)
+			Expect(err).ToNot(HaveOccurred())
+			ctx = settings.ToContext(ctx, &settings.Settings{
+				AssumeRoleARN:           "settings-cluster-role",
+				AssumeRoleDuration:      time.Minute * 22,
+				ClusterCABundle:         "settings-bundle",
+				ClusterName:             "settings-cluster",
+				ClusterEndpoint:         "https://settings-cluster",
+				IsolatedVPC:             true,
+				VMMemoryOverheadPercent: 0.05,
+				InterruptionQueueName:   "settings-cluster",
+				ReservedENIs:            8,
+			})
+			opts.MergeSettings(ctx)
+			expectOptionsEqual(opts, test.Options(test.OptionsFields{
```

Comment:
Should this have been a public `ExpectOptionsEqual` function since we are calling it both in `aws/karpenter-core` and `aws/karpenter`

Reply:
The two structs they're comparing are different, and unless I dive into reflection I don't think it can be made generic.